include/graaflib/algorithm/coloring/welsh_powell.tpp

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <graaflib/algorithm/coloring/welsh_powell.h>
+
+#include <unordered_map>
+#include <vector>
+#include <algorithm>
+#include <iostream>
+
+namespace graaf::algorithm {
+
+template <typename GRAPH>
+std::unordered_map<vertex_id_t, int> welsh_powell_coloring(const GRAPH& graph) {
+    
+    using DegreeVertexPair = std::pair<int, vertex_id_t>;
+
+    // Step 1: Sort vertices by degree in descending order
+    std::vector<DegreeVertexPair> degree_vertex_pairs;
+    for (const auto& vertex : graph.get_vertices()) {
+        int degree = graph.get_degree(vertex);
+        degree_vertex_pairs.emplace_back(degree, vertex);
+    }
+
+    std::sort(degree_vertex_pairs.rbegin(), degree_vertex_pairs.rend());
```

Comment:
[minor]: Such steps of a larger algorithm could be good candidates to pull out into their own function (potentially living in an anonymous namespace).

```c++
template <typename GRAPH>
[[nodiscard]] std::vector<vertex_id_t> compute_sorted_vertices_by_degree(const GRAPH& graph) {
  // the existing code
  // + additional transformation from std::vector<pair> to std::vector<vertex_id_t>
}
```

Reply:
The idea behind writing the entire code in one namespace is that if one wants to check the code he/she would get the steps in serial order and would not have to scroll constantly to look into different functions.Moreover, since this piece of code is not used over and over again converting this into a function would not be as useful as in general cases. Nonetheless, if it's mandatory to do it I'll do it...

Reply:
I am also fine with leaving this as-is, so let's go ahead with the merge.

In general I am an advocate for having a larger number of smaller functions. We can keep them in the same source file (by placing them in the anonymous namespace) and if the functions are well written it makes the high level functions quite nice to read.

Reply:
You do have a valid point but since this is a very small piece of code arranging into functions would not be that beneficial.Thereby let's please go ahead with the merge!!!